app/src/main/java/com/github/sdp_begreen/begreen/FirebaseDB.kt

```diff
@@ -1,38 +1,99 @@
 package com.github.sdp_begreen.begreen
 
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
 import com.google.firebase.database.DatabaseReference
 import com.google.firebase.database.ktx.database
 import com.google.firebase.ktx.Firebase
-import java.util.concurrent.CompletableFuture
+import kotlinx.coroutines.tasks.await
+import java.io.ByteArrayOutputStream
 
 /**
- * Database implementation using Firebase
+ * Database implementation using Firebase's realtime database
  */
-class FirebaseDB : Database() {
+class FirebaseDB() {
 
-    private val firebaseDB: DatabaseReference = Firebase.database.reference
+    /**
+     * Singleton value to access the database
+     */
+    companion object {
+        val db: FirebaseDB = FirebaseDB()
+    }
+
+    private val databaseReference: DatabaseReference = Firebase.database.reference
 
     /**
      * Return the value associated with the given [key]
+     * If an error occurs (by ex: because the key does not exist,
+     * or because the value associated with the key is not a value), returns null
      */
-    override fun get(key: String): CompletableFuture<String> {
+    suspend fun get(key: String): String? {
+        return try {
+            val data = databaseReference.child(key).get().await()
+            data.value?.let { it as? String }
+        } catch (e: Error) {
+            null

```

Comment:
Actually yes, thinking again about it, it might not be the best idea. If we're making a query to Firebase's database, we would like to know whether we got no value because no key was found of because we lost connection. I'll create a new branch to add this fix, aswell as some other changes in the way we store images since it is not optimal for the moment